js/interface/map-panel.js

```diff
@@ -56,7 +131,29 @@ Fliplet.InteractiveMap.component('map-panel', {
         }
       })
 
-      window.filePickerProvider.then((result) => {
+      window.filePickerProvider.then(result => {
+        if (this.shouldKeepMarkers) {
+          let newImageWidth = result.data[0].size[0]
+          let newImageHeight = result.data[0].size[1]
+
+          if (newImageWidth !== this.imageWidth && newImageHeight !== this.imageHeight) {
+            let widthRatioDifference = newImageWidth/this.imageWidth
+            let heightRatioDifference = newImageHeight/this.imageHeight
+
+            this.dataSourceConnection.find().then(records => {
+              records.forEach((elem, index, array) => {
+                if (elem.data['Map name'] === this.name) {
+                  array[index].data['Position X'] *= widthRatioDifference
+                  array[index].data['Position Y'] *= heightRatioDifference
+                }
+              })
+              this.entries = records
+              this.columns = _.keys(records[0].data)
```

Comment:
Do we need to be concerned with what if `records` is empty, i.e. the data source is empty?

Reply:
Same goes here if there are no markers on the map user wants to replace, modal will not appear, code will not execute.